### PR TITLE
Switch to ng-packagr + Use Jest

### PR DESCRIPTION
**Pull request by wmarques**
Made a few changes on the tooling for the library:

- Use ng-packagr, which avoid us to maintain a webpack config and export our lib using the Angular Package Format (not a expert here, but the tool looks really great for building anguar libs)
- Switched to Jest to simplify the config and run test faster
- Switch `ngx-translate` into `peerDepencies` : The usage of dependencies in libraries aren't recommended as this can lead to duplicate dependency versions, so it's better to install `ngx-translate` in the JHipster application instead of ng-jhipster.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,15 +15,12 @@ before_install:
   - export DISPLAY=:99.0
   - sh -e /etc/init.d/xvfb start
   - export CHROME_BIN=chromium-browser
-  # Upgrade NPM
-  - npm install -g npm
```
Comment (pascalgrimaud) [inline .travis.yml]: why removing the upgrade NPM ?

Comment (wmarques) [reply .travis.yml]: If we install node LTS we will have the according NPM version so for me that's not a useful step, it gains some seconds on the build :)
Did you have issues by not upgrading NPM ?

Comment (pascalgrimaud) [reply .travis.yml]: no issue, just consistency for all other builds we have

Comment (wmarques) [reply .travis.yml]: Well i can keep it :)

File: package.json [hunk 2]
```diff
@@ -15,72 +16,68 @@
...
-    "ngc": "ngc",
-    "ngc-aot": "ngc -p tsconfig-aot.json",
-    "build": "npm run cleanup && npm run ngc && webpack && shx cp bundles/ng-jhipster.umd.js bundles/index.js",
-    "build:aot": "npm run cleanup && npm run ngc-aot && webpack && shx cp bundles/ng-jhipster.umd.js bundles/index.js",
-    "build:local": "npm run build && npm test && npm pack",
-    "release": "npm test && git push && git push --tags && npm publish",
+    "test": "jest",
+    "test:watch": "jest --watch",
+    "build": "ng-packagr -p package.json",
+    "release": "npm test && git push && git push --tags && npm publish dist/",
```
[the remark this answers is not in the document]
Comment (deepu105) [reply package.json]: especially for tree shaking and AOT

**Comment by wmarques** [comment]
@deepu105 I'll do some more testing on various apps, but I tried with the sample app, both the dev using `npm start` and the production using `npm run build && ./mvnw` works.
But i'll try some other apps.

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (wmarques) [reply package.json]: Yes but ng-packagr provides the type definitions in the dist directory

[the remark this answers is not in the document]
Comment (deepu105) [reply package.json]: i'm not worried about type definitions, its more about being able to treeshake modules, but if this is the suggested behaviour from the lib then its fine

**Comment by wmarques** [comment]
@deepu105 Tested on a freshly generated app with the dev profile and the prod profile, both works.
Ran through the admin menus and the recently merged metrics page, they work too.
Do you want me to do some tests on sample apps with some entities?

**Comment by deepu105** [comment]
Yes please and if everything is fine go ahead and merge it

Thanks & Regards,
Deepu

**Comment by wmarques** [comment]
@deepu105 
Just tested on the BankApplication sample app, ran protractor, jest, and clicked in every page, works fine 😄
I also checked the production profile, with the AOT, and it works too.
I'm merging this !

**Pull request merged**